### ISSUE TITLE
helm `set` parameter should accept dict

### ISSUE THREAD

**Issue by jazzdan** [tilt-dev/tilt#3558]
For parameters that set a bunch of options, like values in a template, it can be really useful to allow them to be specified as dictionaries. This allows the value being passed in to be safely built up rather than having to do tedious string manipulation.

This would mean that Tilt would be responsible for converting the dictionary in to the string format that the helm set parameter expects.

Originally reported in the #tilt channel:

What I'm looking to do is reuse the values file of our umbrella chart on the individual repos. Example
1. dev installs all services through our umbrella chart
2. dev runs tilt up in a specific repo
I want (2) to read the values file of the umbrella chart and apply both the global section, and extract the contents of the chart-specific values.
Example of global `values.yaml`:
```
global:
  db:
    username: "test"
    password: "password"
    host: "mysql"
# configure a subchart
notifications:
  channels:
    slack:
      webhookUrl: "https://..."
```
When I run tilt up in the notifications project, I want to pass the following to the helm function:
```
global:
  db:
    username: "test"
    password: "password"
    host: "mysql"
channels:
  slack:
    webhookUrl: "https://..."
```
basically extracting the values for that particular chart. I can do the extraction just file with read_yaml, but I don't think there's a way for me to then pass this into the helm function? Is there a way for me to work around this? One workaround I've considered is converting the nested yaml syntax into the dot-nested syntax that --set accepts, but it seems a bit tedious

**Comment by nicks** [comment]
I also commented on this on the thread, but i usually see people do this with the `values=` argument. People have also requested `--set-file` https://github.com/tilt-dev/tilt/issues/2752. 

I'm a little bit nervous about adding a way to pass in params that's not natively supported in Helm, given the sheer number of APIs for this they offer already

**Comment by phroggyy** [comment]
@nicks as I was the one to request this I think it's only fair I provide alternatives: what I _actually_ want is to apply yaml. I just want to programmatically modify it first. One approach we could take to this would thus be to write the dict to a yaml file (which is already supported through `encode_yaml` and then writing to a file). The perfect thing would be if helm would accept stdin for values file contents, but I think Tilt writing to a file behind the scenes is a much more realistic thing.

From the user perspective, what I want to avoid is just the boilerplate around creating the file, making sure it's deleted when I stop tilt, etc.
